packages/firebase_admob/android/src/main/java/io/flutter/plugins/firebaseadmob/MobileAd.java

```diff
@@ -226,4 +248,64 @@ void show() {
 
     // It is not possible to hide/remove/destroy an AdMob interstitial Ad.
   }
+
+  static class Native extends MobileAd {
+    private UnifiedNativeAd nativeAd;
+    private final Map<String, Object> customOptions;
+    private final NativeAdFactory nativeAdFactory;
+
+    private Native(
+        int id,
+        Activity activity,
+        MethodChannel channel,
+        NativeAdFactory nativeAdFactory,
+        Map<String, Object> customOptions) {
+      super(id, activity, channel);
+      this.nativeAdFactory = nativeAdFactory;
+      this.customOptions = customOptions;
+    }
+
+    @Override
+    void load(String adUnitId, Map<String, Object> targetingInfo) {
+      status = Status.LOADING;
+
+      final AdLoader adLoader =
+          new AdLoader.Builder(activity, adUnitId)
+              .forUnifiedNativeAd(
+                  new UnifiedNativeAd.OnUnifiedNativeAdLoadedListener() {
+                    @Override
+                    public void onUnifiedNativeAdLoaded(UnifiedNativeAd unifiedNativeAd) {
+                      nativeAd = unifiedNativeAd;
+                      onAdLoaded();
+                    }
+                  })
+              .withAdListener(this)
+              .withNativeAdOptions(new NativeAdOptions.Builder().build())
+              .build();
+
+      AdRequestBuilderFactory factory = new AdRequestBuilderFactory(targetingInfo);
+      adLoader.loadAd(factory.createAdRequestBuilder().build());
+    }
+
+    @Override
+    void show() {
+      if (status == Status.LOADING) {
```

Comment:
Seems like this is already done in showAdView

Reply:
I created an abstract class called `MobileAdWithView` that `Banner` and `NativeAd` extend.